### ISSUE TITLE
false positive redundant cast

### ISSUE THREAD

**Issue by MkazemAkhgary** [dotnet/roslyn#32773]
Assuming that bounds is long, start and end is int. following gives a warning

>Bitwise-or operator used on a sign-extended operand; consider casting to a smaller unsigned type first

```c#
bounds = (((long)end) << 32) | start;
```

In my case it doesn't really matter to use sign or zero extension. start is always positive. for sake of hiding the warning

```c#
bounds = (((long)end) << 32) | ((long)start);
```

says cast is redundant. I think if compiler suggests to resolve ambiguity, then it shouldn't suggest to introduce ambiguity again. (in my case it doesn't matter but i thought its good to share)

**Comment by gafter** [comment]
Moving to IDE as it is not the compiler that says the cast is redundant.
